src/tests/ftest/checksum/basic_checksum.py

```diff
@@ -0,0 +1,178 @@
+"""
+
+import os
+import ctypes
+import time
+import agent_utils
+import server_utils
+import write_host_file
+
+from pydaos.raw import (DaosPool, DaosContainer, IORequest,
+                        DaosObj)
+from apricot import TestWithServers
+
+
+class ChecksumContainerValidation(TestWithServers):
+    """
+    Test Class Description:
+
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-instance-attributes
+    def setUp(self):
+        super(ChecksumContainerValidation, self).setUp()
+        self.agent_sessions = None
+        self.pool = None
+        self.container = None
+        self.obj = None
+        self.ioreq = None
+        self.hostlist = None
+        self.hostfile = None
+        self.no_of_dkeys = None
+        self.no_of_akeys = None
+        self.array_size = None
+        self.record_length = None
+        server_group = self.params.get("name",
+                                       '/server_config/',
+                                       'daos_server')
+        self.hostlist = self.params.get("test_servers", '/run/hosts/*')
+        self.hostfile = write_host_file.write_host_file(self.hostlist,
+                                                        self.workdir)
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')
+
+        self.agent_sessions = agent_utils.run_agent(
+            self, self.hostlist)
+        server_utils.run_server(self, self.hostfile, server_group)
+
+        self.pool = DaosPool(self.context)
+        self.pool.create(self.params.get("mode", '/run/pool/createmode/*'),
+                         os.geteuid(),
+                         os.getegid(),
+                         self.params.get("size", '/run/pool/createsize/*'),
+                         self.params.get("setname", '/run/pool/createset/*'),
+                         None)
+        self.pool.connect(2)
+        self.csum = self.params.get("enable_checksum", '/run/container/*')
+        self.container = DaosContainer(self.context)
+        self.container.create(poh=self.pool.handle, enable_chksum=self.csum)
+        self.container.open()
+
+        self.obj = DaosObj(self.context, self.container)
+        self.obj.create(objcls=1)
+        self.obj.open()
+        self.ioreq = IORequest(self.context,
+                               self.container,
+                               self.obj, objtype=4)
+
+    def tearDown(self):
+        try:
+            if self.container:
+                self.container.close()
+                self.container.destroy()
+            if self.pool:
+                self.pool.disconnect()
+                self.pool.destroy(1)
+        finally:
+            if self.agent_sessions:
+                agent_utils.stop_agent(self.agent_sessions)
+            server_utils.stop_server(hosts=self.hostlist)
```

Comment:
No need to override tearDown(). All the items inside will be taken care by apricots tearDown()

Reply:
Will do. Thanks.